cmd/zpool/zpool_main.c

```diff
@@ -2163,19 +2173,84 @@ show_import(nvlist_t *config)
 	}
 }
 
+typedef long long unsigned int llu;
+
+static void
+extract_timestamp_and_txg(nvlist_t *config, uint64_t *txg,
+    uint64_t *timestamp)
+{
+	verify(nvlist_lookup_uint64(config, ZPOOL_CONFIG_TIMESTAMP,
+	    timestamp) == 0);
+	verify(nvlist_lookup_uint64(config, ZPOOL_CONFIG_FOUND_TXG,
+	    txg) == 0);
+}
+
+static int
+check_pool_for_activity(libzfs_handle_t *hdl, nvlist_t *config)
+{
+	nvlist_t *nvl = NULL;
+	uint64_t original_txg, original_timestamp;
+	int rc = 0;
+	char *poolname;
+	int seconds_between_checks = 1;
+	int times_to_check = 10;
```

Comment:
@richardelling thanks for the feedback.

I tend to agree we're going to want to provide a few options, since it's going to be difficult to pick values which work for the wide variety of hardware and pool configurations possible.  However, I think we can settle on reasonable defaults which work well for vast majority of systems.

The current design includes a `zfs_mmp_interval` module option which controls the frequency of the mmp writes.  You'll be able to increase this, or something similar, to reduce the frequency of writes for limited-endurance media.  For large pools we expect this to be less of an issue since the updates will be spread randomly over the vdevs.  But for single disk pools I can understand the concern.

On the `zpool import` side of things where we're only reading the uberblocks I think it's reasonable to provide someway to specify an increased wait time.  Although, I agree with @ofaaland we need to enforce some reasonable minimum.

> temporary, transient conditions like HBA or drive resets are on the order of 10 to 15 seconds.

There are admittedly entire classes of IO related hiccups which can cause problems.